code/modules/mechs/components/_components.dm

```diff
@@ -101,27 +109,91 @@
-	if(I.use_tool(user, src, WORKTIME_INSTANT, QUALITY_SCREW_DRIVING, FAILCHANCE_ZERO))
-		if(contents.len)
-			//Filter non movables
-			var/list/valid_contents = list()
-			for(var/atom/movable/A in contents)
-				if(!A.anchored)
-					valid_contents += A
-			if(!valid_contents.len)
-				return
-			var/obj/item/removed = pick(valid_contents)
-			if(!(removed in contents))
-				return
-			if(eject_item(removed, user))
-				update_components()
-		else
-			to_chat(user, SPAN_WARNING("There is nothing to remove."))
+	var/list/usable_qualities = list(QUALITY_PULSING, QUALITY_BOLT_TURNING, QUALITY_WELDING, QUALITY_PRYING, QUALITY_SCREW_DRIVING)
+	var/tool_type = I.get_tool_type(user, usable_qualities, src)
+
+	switch(tool_type)
+		if(QUALITY_SCREW_DRIVING)
+			if(contents.len)
+				//Filter non movables
+				var/list/valid_contents = list()
+				for(var/atom/movable/A in contents)
+					if(!A.anchored)
+						valid_contents += A
+				if(!valid_contents.len)
+					return
+				var/obj/item/removed = pick(valid_contents)
+				if(!(removed in contents))
+					return
+				if(eject_item(removed, user))
+					update_components()
+			else
+				to_chat(user, SPAN_WARNING("There is nothing to remove."))
+
+		if(QUALITY_WELDING)
+			repair_brute_generic(I, user)
+			return
+
+	if(istype(I, /obj/item/stack/cable_coil))
+		repair_burn_generic(I, user)
 		return
+
 	if(istype(I, /obj/item/device/robotanalyzer))
 		to_chat(user, SPAN_NOTICE("Diagnostic Report for \the [src]:"))
 		return_diagnostics(user)
 	return ..()
 
 /obj/item/mech_component/proc/update_components()
 	return
+
+
+/obj/item/mech_component/proc/repair_brute_generic(obj/item/I, mob/user)
+	var/weld_amt = round(max(0, 10, 5 + user.stats.getStat(STAT_MEC) / 3))
+
+	if(brute_damage <= 0)
+		to_chat(user, SPAN_NOTICE("You inspect \the [src] but find nothing to weld."))
+		return
+
+	if(total_damage >= max_damage)
+		to_chat(user, SPAN_WARNING("This part is too damaged to be repaired with a welder!"))
+		return
+
+	if(QUALITY_WELDING in I.tool_qualities)
+		if(I.use_tool(user, src, WORKTIME_NORMAL, QUALITY_WELDING, FAILCHANCE_NORMAL, required_stat = STAT_MEC))
+			visible_message(SPAN_WARNING("\The [src] has been repaired by [user]!"),"You hear welding.")
+			repair_brute_damage(weld_amt)
+			return
+
+/obj/item/mech_component/proc/repair_burn_generic(obj/item/stack/cable_coil/CC, mob/user)
+	var/wire_amt = round(max(10, 5 + user.stats.getStat(STAT_MEC) / 3))
+
+	if(burn_damage <= 0)
+		to_chat(user, SPAN_NOTICE("You inspect /the [src]'s wiring, but can't find anything to fix."))
+		return
+
+	if(CC.amount < 5)
+		to_chat(user, SPAN_WARNING("You need at least 5 cable coil pieces in order to replace wiring."))
+		return
+
+	if(total_damage >= max_damage)
+		to_chat(user, SPAN_WARNING("This part is too damaged to be re-wired!"))
+		return
+
+	to_chat(user, SPAN_NOTICE("You start replacing wiring in \the [src]."))
+
+	if(do_mob(user, src, 30) && CC.use(5))
```

Comment:
```suggestion
	if(do_mob(user, src, 30) && CC.use(5))
		user.visible_message(SPAN_NOTICE("\The [src] has been rewired by [user]!"), SPAN_NOTICE("You rewired \the [src]."))
```

Reply:
I feel like this'd spam the chat when you're repairing lots of damage

Reply:
Not really a code policy, it was just to provide a bit of feedback to other people so that they know that this dude is not standing in front of the mech without doing anything.

Reply:
That's fair! I think I made it so there's a finisher message "you wire the x" but no starter, so you can't just go like:

Get damaged part > Wire it > Walk away, cancelling the do after > Repeat (to be annoying)